### PR TITLE
Simplify code for several old template tags.

### PR DESCRIPTION
**Pull request by solarissmoke**
Cleans up several old template tags to considerably simplify the code (at the time they were written Django's template tag framework was not as easy to use). None of these tags had any tests, so I've added tests for each as well.

### CODE REVIEW

File: src/oscar/templatetags/wishlist_tags.py [hunk 1]
```diff
@@ -3,36 +3,10 @@
 register = template.Library()
 
 
-@register.tag(name="wishlists_containing_product")
-def do_basket_form(parse, token):
+@register.simple_tag
+def wishlists_containing_product(wishlists, product):
     """
     Template tag for adding the user's wishlists form to the
     template context so it can be rendered.
```
Comment (jwayodi) [inline src/oscar/templatetags/wishlist_tags.py:10]: Please remove the bit in the docstring suggesting it adds the variable to the template context (which was an implementation detail of the previous tag), and just say it returns the value? Or just remove the whole thing.

File: src/oscar/templatetags/sorting_tags.py [hunk 2]
```diff
@@ -21,45 +22,27 @@
...
+        direction = get_vars.pop('dir', [''])[0]
+        get_vars['dir'] = sort_directions[direction]['inverse']
+        icon = sort_directions[direction]['icon']
+
+    href = '%s?sort=%s' % (request.path, field)
+    if len(get_vars) > 0:
+        href += "&%s" % get_vars.urlencode()
+    if icon:
+        title = "%s %s" % (title, icon)
+    return mark_safe('<a href="%s">%s</a>' % (href, title))
```
Comment (jwayodi) [inline src/oscar/templatetags/sorting_tags.py:48]: It's hard to imagine, but could we somehow accidentally end up with an unsafe `title`?

Comment (solarissmoke) [reply src/oscar/templatetags/sorting_tags.py:48]: Seems unlikely, but I've added escaping for it (and included a check in the tests).

**Review by jwayodi — APPROVED** [review]
This looks good.

**Pull request merged**